app/src/main/java/io/gnosis/safe/ui/transactions/details/view/AmountView.kt

```diff
@@ -0,0 +1,70 @@
+package io.gnosis.safe.ui.transactions.details.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.constraintlayout.widget.ConstraintLayout
+import io.gnosis.data.backend.dto.TransactionDirection
+import io.gnosis.data.models.TransactionInfo
+import io.gnosis.data.models.TransferInfo
+import io.gnosis.safe.databinding.ViewTxAmountBinding
+import io.gnosis.safe.utils.formatAmount
+import io.gnosis.safe.utils.loadTokenLogo
+import java.math.BigInteger
+
+class AmountView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private val binding by lazy { ViewTxAmountBinding.inflate(LayoutInflater.from(context), this) }
+
+    fun setAmount(txInfo: TransactionInfo) {
+        when (txInfo) {
+            is TransactionInfo.Custom -> binding.amountTitle.text = txInfo.value.formatAmount(true, 18, "ETH")
+            is TransactionInfo.Transfer -> {
+                val incoming = txInfo.direction == TransactionDirection.INCOMING
+                val decimals: Int = when (val transferInfo = txInfo.transferInfo) {
+                    is TransferInfo.Erc20Transfer -> {
+                        transferInfo.decimals ?: 0
+                    }
+                    is TransferInfo.EtherTransfer -> 18
+                    else -> 0
+                }
+                val symbol: String = when (val transferInfo = txInfo.transferInfo) {
+                    is TransferInfo.Erc20Transfer -> {
+                        transferInfo.tokenSymbol ?: ""
+                    }
+                    is TransferInfo.Erc721Transfer -> {
+                        transferInfo.tokenSymbol ?: ""
+                    }
+                    else -> {
+                        "ETH"
+                    }
+                }
+                binding.amountTitle.text = txInfo.transferInfo.value()?.formatAmount(incoming, decimals, symbol)
+
+                val logoUri = when (val transferInfo = txInfo.transferInfo) {
+                    is TransferInfo.Erc20Transfer -> {
+                        transferInfo.logoUri
+                    }
+                    is TransferInfo.Erc721Transfer -> {
+                        transferInfo.logoUri
+                    }
+                    else -> {
+                        "local::ethereum"
+                    }
+                }
+                binding.logo.loadTokenLogo(logoUri)
```

Comment:
There is a lot of code duplication here that I think could be avoided, if we added a method in `TransferInfo` where we can do the type check and return the correct field. Not terribly important though, but then we could really take advantage of polymorphism.

Reply:
This `Amount` view should be as "dumb" as possible, in a sense that it receives an amount and logo to display. Maybe that would simplify things a bit.

Reply:
I totally agree. The View should be as dumb as possible. If only because it cannot be unit tested. I don't know why I hesitated to put helper methods into TransferInfo/TransactionInfo